ksqldb-engine/src/main/java/io/confluent/ksql/schema/ksql/inference/SchemaRegistryTopicSchemaSupplier.java

```diff
@@ -124,7 +127,19 @@ private SchemaResult fromParsedSchema(
       final boolean isKey
   ) {
     final Format format = formatFactory.apply(expectedFormat);
-    final SchemaTranslator translator = format.getSchemaTranslator(expectedFormat.getProperties());
+
+    // Put schema id to properties so that translator will not uppercase field name
+    boolean schemaIdInProperty =
+        isKey ? expectedFormat.getProperties().containsKey(ConnectFormat.KEY_SCHEMA_ID)
+            : expectedFormat.getProperties().containsKey(ConnectFormat.VALUE_SCHEMA_ID);
+    final Map<String, String> propertiesWithId = schemaIdInProperty ? expectedFormat.getProperties()
+        : new ImmutableMap.Builder<String, String>().putAll(
+                expectedFormat.getProperties())
+            .put(isKey ? ConnectFormat.KEY_SCHEMA_ID : ConnectFormat.VALUE_SCHEMA_ID,
+                String.valueOf(id))
```

Comment:
Does this mean we're changing the default behavior (even when no explicit schema ID is supplied in the WITH clause properties) to no longer uppercase field names? Was this discussed as part of the original KLIP? I was under the impression that we were going to leave the default behavior as is and allow users to explicitly specify SCHEMA_ID in the WITH clause properties if they wanted to preserve casing, but maybe I've misunderstood.

Reply:
Even if there are no backwards compatibility concerns for statements that have already been written to the command topic, the change is still backwards incompatible in that the same (new) statement issued on different server versions will produce different results. Working with streams with lowercase field names in ksqlDB is not very intuitive/convenient: the field names must be escaped in all subsequent statements. This is a rather large change for users to adjust to. That's not to say that we shouldn't make it, just that I think it warrants justification in a KLIP and I'd want someone more user-facing onboard cc @colinhicks

Reply:
Sounds good. @colinhicks please take a look whether we want to have this behavior for schema inference